bndtools.builder/src/org/bndtools/builder/BndtoolsBuilder.java

```diff
@@ -113,16 +118,57 @@
 					.getDynamicReferences();
 			}
 
-			if (model == null) {
-				try {
-					model = Central.getProject(myProject);
-				} catch (Exception e) {
-					markers.deleteMarkers("*");
+			Project ourModel = null;
+			try {
+				ourModel = Central.getProject(myProject);
+			} catch (Exception e) {
+				markers.deleteMarkers("*");
+				// If the cnf project is missing there will likely be a myriad
+				// of Java problems. These do nothing but clutter the error
+				// output and make it hard to find the real problem in the
+				// problem log.
+				myProject.deleteMarkers(IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER, true, IResource.DEPTH_INFINITE);
+				logger.logError("Exception while trying to fetch bnd project for " + myProject.getName(), e);
+				markers.createMarker(null, IMarker.SEVERITY_ERROR, "Exception while trying to fetch bnd project: " + e,
+					BndtoolsConstants.MARKER_BND_PATH_PROBLEM);
+				return noreport();
+			}
+			if (ourModel == null) {
+				markers.deleteMarkers("*");
+				// If the cnf project is missing there will likely be a myriad
+				// of Java problems. These do nothing but clutter the error
+				// output and make it hard to find the real problem in the
+				// problem log.
+				myProject.deleteMarkers(IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER, true, IResource.DEPTH_INFINITE);
```

Comment:
It does seem a bit presumptuous of us (Bndtools) to remove the error markers of another builder, no? I am not sure we should do this.